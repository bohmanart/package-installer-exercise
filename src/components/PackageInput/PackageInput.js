import React from 'react'

const PackageInput = (props) => (
    <div>
        <h2>Input</h2>
        <pre><code>"
            {props.input}
        "</code></pre>
    </div>
)

export default PackageInput
