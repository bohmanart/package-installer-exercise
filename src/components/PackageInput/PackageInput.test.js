import React from 'react'
import renderer from 'react-test-renderer'
import PackageInput from './PackageInput'

describe('<PackageInput/>', () => {
    
    it('should render `h2`, `pre` and `code` elements', () => {
        const tree = renderer.create(
            <PackageInput/>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('should render a code example of an array with package names and dependencies from an array passed in', () => {
        const input = [
            "KittenService: ",
            "Leetmeme: Cyberportal",
            "Cyberportal: Ice",
            "CamelCaser: KittenService",
            "Fraudstream: Leetmeme",
            "Ice: "
        ]

        const tree = renderer.create(
            <PackageInput input={input}/>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })

})
