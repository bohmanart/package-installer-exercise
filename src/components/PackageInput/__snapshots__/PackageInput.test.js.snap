// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PackageInput/> should render \`h2\`, \`pre\` and \`code\` elements 1`] = `
<div>
  <h2>
    Input
  </h2>
  <pre>
    <code>
      [
      <br />
        "
      "
      <br />
      ]
    </code>
  </pre>
</div>
`;

exports[`<PackageInput/> should render a code example of an array with package names and dependencies from an array passed in 1`] = `
<div>
  <h2>
    Input
  </h2>
  <pre>
    <code>
      [
      <br />
        "
      KittenService: 
      Leetmeme: Cyberportal
      Cyberportal: Ice
      CamelCaser: KittenService
      Fraudstream: Leetmeme
      Ice: 
      "
      <br />
      ]
    </code>
  </pre>
</div>
`;
